smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -188,6 +188,38 @@ contract MixinTransfer is
     return fee.mul(transferFeeBasisPoints) / BASIS_POINTS_DEN;
   }
 
+  /**
+  * @notice Modify the expirationTimestamp of a key
+  * by a given amount.
+  * @param _tokenId The ID of the key to modify.
+  * @param _deltaT The amount of time in seconds by which
+  * to modify the keys expirationTimestamp
+  * @param _addTime Choose whether to increase or decrease
+  * expirationTimestamp (false == decrease, true == increase)
+  * @dev Throws if owner does not have a valid key.
+  */
+  function _timeMachine(
+    uint _tokenId,
+    uint256 _deltaT,
+    bool _addTime
+  ) internal
+  {
+    address tokenOwner = ownerOf[_tokenId];
+    require(tokenOwner != address(0), 'NON_EXISTENT_KEY');
+    Key storage key = keyByOwner[tokenOwner];
+    uint formerTimestamp = key.expirationTimestamp;
+    bool validKey = getHasValidKey(tokenOwner);
+    if(_addTime) {
+      if(validKey) {
+        key.expirationTimestamp = formerTimestamp.add(_deltaT);
+      } else {
+        key.expirationTimestamp = block.timestamp.add(_deltaT);
+      }
+    } else {
+      key.expirationTimestamp = formerTimestamp - _deltaT;
```

Comment:
Should we maybe use `.sub` here?

Reply:
absolutely. Ive been staring at this thing too long ;)